src/js/sanitize.js

```diff
@@ -163,11 +163,17 @@ const RECIPE_QUANTITY_RE = new RegExp([
 function sanitizeIngredient (ingredient) {
   const match = ingredient.match(RECIPE_QUANTITY_RE);
 
-  if (match === null) {
-    return {ingredient};
+  // only set properties for the capture groups which ended up matching.
```

Comment:
Does this change anything downstream compared to having them set to `null`?

Reply:
It is really hard to know because javascript. JavaScript gives you ways to treat various values as "truthy" or "falsey", and it also gives you ways to explicitly distinguish between `false`, `null`, *empty string*, *empty array*, `x: undefined`, and *x is not present in y*. I generally prefer `if (x) ...` to `if (x === null)` for this exact reason, but occasionally the distinction matters. Are you doing anything crazy elsewhere where the difference would matter?

The only code that *should* explicitly about the above distinction is the test framework, and rightfully so. I've made adjustments to a couple test cases.

This way of doing it resulted only one test case needing to be changed, vs several of them.

Reply:
> Are you doing anything crazy

probably, I use javascript only begrudgingly. 

It should be easy enough to confirm, it's a small project. As long as the recipe view continues to render when these values are `undefined` instead of `null` this should be fine

Reply:
From what I've seen, though I didn't explicitly look for misbehavior in this case so it's possible I missed it. So I guess I should add that to my manual testing.